### ISSUE TITLE
Plugin import breaks if CMAKE_RUNTIME_OUTPUT_DIRECTORY_* is set

### ISSUE THREAD

**Issue by Gollum999** [mosra/magnum#486]
### Environment
* Windows 10
* Visual Studio 16.8.2 (MSVC 19.28.29334)
* CMake 3.18.1
* Magnum mosra/magnum@92d0609e33252c4e1f9e772dfb0825e96122aba3
* Corrade mosra/corrade@e5970c4190717a6425f1b17241f19a62341569ef

### Problem
If `CMAKE_RUNTIME_OUTPUT_DIRECTORY_<CONFIG>` is set, trying to load plugins will fail.  Example error message:

```
PluginManager::Manager::Manager(): none of the plugin search paths in {C:/code/magnum/bld/vs/Debug/bin/magnum-d/importers, magnum-d/importers} exists and pluginDirectory was not set, skipping plugin discovery
PluginManager::Manager::load(): plugin TgaImporter is not static and was not found in
```
As described [here](https://cmake.org/cmake/help/latest/variable/CMAKE_RUNTIME_OUTPUT_DIRECTORY_CONFIG.html#cmake-runtime-output-directory-config), if this variable is set, it is used to initialize the [`RUNTIME_OUTPUT_DIRECTORY_<CONFIG>`](https://cmake.org/cmake/help/latest/prop_tgt/RUNTIME_OUTPUT_DIRECTORY_CONFIG.html#runtime-output-directory-config) property of each target.  This property takes precedence over the [`RUNITME_OUPUT_DIRECTORY`](https://cmake.org/cmake/help/latest/prop_tgt/RUNTIME_OUTPUT_DIRECTORY.html#runtime-output-directory) property.  This means that the output directories that are set in each plugin's [CMakeLists](https://github.com/mosra/magnum/blob/92d0609e33252c4e1f9e772dfb0825e96122aba3/src/MagnumPlugins/TgaImporter/CMakeLists.txt#L50) end up being overridden.  This causes the plugin library to not be built into the expected location, causing it to fail to load at runtime.

### To reproduce
Build Magnum as a subproject as in [Example 1a](https://doc.magnum.graphics/magnum/getting-started.html#getting-started-setup-subproject) of the Getting Started guide.  Add the following lines to CMake before `add_subdirectory(magnum)`:

```
set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
```
Then build with some plugin enabled and try to load the plugin at runtime.

**Comment by Gollum999** [comment]
Just as a note, in my case I ran into this because of [Conan](https://conan.io/).  By default, [`conan_basic_setup`](https://docs.conan.io/en/latest/reference/generators/cmake.html#conan-basic-setup) modifies the build output directories by setting these CMake variables.  I switched to using `conan_basic_setup(NO_OUTPUT_DIRS)` to avoid this behavior.

**Comment by mosra** [comment]
Hi,

yeah, the plugins are expected to be in those particular subdirectories (`magnum/blah`) relative to the interface library, otherwise it's not really possible for the plugin manager to discover them or know which plugin is which. The library already decides based on presence / value of `CMAKE_RUNTIME_OUTPUT_DIRECTORY` on its own, but isn't currently aware of `CMAKE_RUNTIME_OUTPUT_DIRECTORY_<CONFIG>`. 

I'll have to fix that.

**Comment by mosra** [comment]
This should be fixed with #570. Sorry it took half a century to get this done.
